File: 1516-the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n.cpp
class Solution {
    string ans;
    vector<char> dex{'a', 'b', 'c'};

    int bkg(int n, int k, string dp) {
        if (dp.size() == n) {
            if (--k == 0) {
                ans = dp;
                return 0;
            }

            return k;
        }

        for (auto c : dex) {
            if (dp.empty() || dp.back() != c) {
                k = bkg(n, k, dp + c);
                if (!k)
                    return 0;
            }
        }
        return k;
    }

public:
    string getHappyString(int n, int k) { return bkg(n, k, "") ? "" : ans; }
};

File: 1516-the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        q = deque(["a", "b", "c"])
        ans = []

        while q:
            s = q.popleft()
            if len(s) == n:
                ans.append(s)
                continue
            for c in "abc":
                if s[-1] != c:
                    q.append(s + c)
        return "" if k > len(ans) else ans[k-1]

        